opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/NodeDaoHibernate.java

```diff
@@ -554,4 +555,22 @@ private OnmsNode getNode(Integer nodeId, Session session) {
         return null;
     }
 
+
+
+    public List<OnmsNode> findNodeWithMetaData(final String context, final String key, final String value) {
+
+        return getHibernateTemplate().execute(session -> (List<OnmsNode>) session.createSQLQuery("SELECT n.nodeid FROM node n, node_metadata m WHERE m.id = n.nodeid AND context = '" + StringEscapeUtils.escapeSql(context) + "' AND key = '" + StringEscapeUtils.escapeSql(key) + "' AND value = '" + StringEscapeUtils.escapeSql(value) + "'")
```

Comment:
Use `SQLQuery#setString` instead of `StringEscapeUtils`.

Reply:
done